### PR TITLE
support shared storage for reusable mode

### PR DESCRIPTION
**Pull request by J-shang**
[storage]
- [x] nfs
- [ ] azure file
- [x] azure blob

[training service]
- [x] local
- [x] remote
- [x] pai
- [x] aml

[others]
- [ ] test
- [x] example
- [x] doc

### CODE REVIEW

File: ts/nni_manager/training_service/reusable/shared_storages/azureblobStorageService.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    wget https://packages.microsoft.com/config/ubuntu/$version/packages-microsoft-prod.deb
+    sudo dpkg -i packages-microsoft-prod.deb
+    sudo apt-get update
+    sudo apt-get install -y blobfuse fuse
+elif [ $id = "CentOS" ] || [ $id = "RHEL" ]
+then
+    sudo rpm -Uvh https://packages.microsoft.com/config/rhel/$(echo $version | cut -c1)/packages-microsoft-prod.rpm
+    sudo yum install -y blobfuse fuse
+else
+    echo "Not support distributor."
```
Comment (liuzhe-lz) [inline ts/nni_manager/training_service/reusable/shared_storages/azureblobStorageService.ts:48]: I feel uncomfortable about this...

Comment (liuzhe-lz) [inline ts/nni_manager/training_service/reusable/shared_storages/azureblobStorageService.ts:48]: I feel uncomfortable about this...

File: docs/en_US/Tutorial/HowToUseSharedStorage.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
+**How to Use Shared Storage**
```
Comment (liuzhe-lz) [inline docs/en_US/Tutorial/HowToUseSharedStorage.rst:1]: Please explicitly list supported distros and state that it requires root privilege.

Comment (J-shang) [reply docs/en_US/Tutorial/HowToUseSharedStorage.rst:1]: add a note below.

**Review by liuzhe-lz — APPROVED** [review]

File: ts/nni_manager/training_service/reusable/shared_storages/nfsStorageService.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+import { MountedStorageService } from '../storages/mountedStorageService';
+import { TrialConfigMetadataKey } from '../../common/trialConfigMetadataKey';
+import { getLogger, Logger } from '../../../common/log';
+import { getExperimentId } from '../../../common/experimentStartupInfo';
+
+const INSTALL_NFS_CLIENT = `
+#!/bin/bash
+if [ -n "$(command -v apt-get)" ]
+then
+    sudo apt-get update
```
Comment (SparkSnail) [inline ts/nni_manager/training_service/reusable/shared_storages/nfsStorageService.ts:19]: If user's os already installed NFS, these bash script does not need to be executed.

File: docs/en_US/Tutorial/HowToUseSharedStorage.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    sharedStorage:
+        storageType: AzureBlob
+        localMountPoint: ${your/local/mount/point}
+        remoteMountPoint: ${your/remote/mount/point}
+        storageAccountName: ${replace_to_your_storageAccountName}
+        storageAccountKey: ${replace_to_your_storageAccountKey}
+        # If you did not set storageAccountKey, you need use `az login` with Azure CLI at first and set resourceGroupName.
+        # resourceGroupName: ${replace_to_your_resourceGroupName}
+        containerName: ${replace_to_your_containerName}
```
Comment (SparkSnail) [inline docs/en_US/Tutorial/HowToUseSharedStorage.rst:30]: What does containerName mean? Is it a contain id?

File: docs/en_US/Tutorial/HowToUseSharedStorage.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    Shared storage is currently in the experimental stage. We suggest use AzureBlob under Ubuntu/CentOS/RHEL, and NFS under Ubuntu/CentOS/RHEL/Fedora/Debian for remote.
+    And make sure your local machine can mount NFS or fuse AzureBlob and has ``sudo`` permission on your remote runtime. We only support shared storage under training service with reuse mode for now.
+
+Example
+-------
+If you want to use AzureBlob, add below to your config. Full config file see :githublink:`mnist-sharedstorage/config_azureblob.yml <examples/trials/mnist-sharedstorage/config_azureblob.yml>`.
+
+.. code-block:: yaml
+
+    sharedStorage:
```
Comment (SparkSnail) [inline docs/en_US/Tutorial/HowToUseSharedStorage.rst:22]: add doc for these new added configuration in ExperimentConfig.rst?

File: docs/en_US/Tutorial/HowToUseSharedStorage.rst (hunk 4 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToUseSharedStorage.rst:30]: The name of the container in blob service in your storage account, like `nni`.
image

File: ts/nni_manager/training_service/reusable/shared_storages/nfsStorageService.ts (hunk 3 above)
Comment (J-shang) [reply ts/nni_manager/training_service/reusable/shared_storages/nfsStorageService.ts:24]: Add a `if [ -n "$(command -v nfsstat)" ]` judgment

File: docs/en_US/Tutorial/HowToUseSharedStorage.rst (hunk 5 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToUseSharedStorage.rst:22]: add them

**Review by SparkSnail — APPROVED** [review]

**Pull request merged**